### DISCUSSION TITLE
PathLayer and FillStyleExtension

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#5591] [Q&A]
Hello all,

It seems like that FillStyleExtension does not work with PathLayer. 

From the document: https://deck.gl/docs/api-reference/extensions/fill-style-extension

> The FillStyleExtension adds selected features to layers that render a "fill", such as the PolygonLayer and ScatterplotLayer.

Can FillStyleExtension be used with PathLayer? If not, is it possible to add that to PathLayer?

Thank you!

**Comment by kylebarron** [comment]
The `FillStyleExtension` renders a pattern in an area. Lines don't have area, therefore you can't use a fill style on a `LineLayer`. The PathStyleExtension is intended for lines. https://deck.gl/docs/api-reference/extensions/path-style-extension

**Comment by TextTL** [comment]
Is it possible to write an extension to create stripes like this for PathLayer?  (I need to know before diving into it). 

![2021-03-18](https://user-images.githubusercontent.com/71099296/111731062-ec0c4680-882f-11eb-84ff-fc546eff6fc9.png)

Thanks,

**Reply by kylebarron** [reply]
You'd likely have to write your own extension shader, but it's likely possible. You can look into the PathStyleExtension source code for inspiration.

**Reply by TextTL** [reply]
Thanks for the guidance!
